Shared/Tests.Shared/AddOrUpdateTests.cs

```diff
@@ -846,6 +846,86 @@ public void AddOrUpdate_WhenObjectHasNonEmptyList_ShouldNotThrow()
             }, Throws.Nothing);
         }
 
+        [Test]
+        public void AddOrUpdate_WhenListIsNull_ShouldNotThrow()
+        {
+            var first = new PrimaryKeyWithPKList
+            {
+                Id = 42
+            };
+
+            Assert.That(() =>
+            {
+                _realm.Write(() => _realm.Add(first, update: true));
+            }, Throws.Nothing);
+        }
+
+        [Test]
+        public void AddOrUpdate_WhenListObjectsHavePK_ShouldOverwriteList()
+        {
+            var first = new PrimaryKeyWithPKList
+            {
+                Id = 1
+            };
+
+            first.ListValue.Add(new PrimaryKeyObject
+            {
+                Id = 1
+            });
+
+            _realm.Write(() => _realm.Add(first));
+
+            var second = new PrimaryKeyWithPKList
+            {
+                Id = 1
+            };
+
+            second.ListValue.Add(new PrimaryKeyObject
+            {
+                Id = 2
+            });
+
+            _realm.Write(() => _realm.Add(second, update: true));
+
+            Assert.That(first.ListValue, Is.EquivalentTo(second.ListValue));
+            Assert.That(first.ListValue.Count, Is.EqualTo(1));
+            Assert.That(first.ListValue[0].Id, Is.EqualTo(2));
+            Assert.That(_realm.All<PrimaryKeyObject>().Count(), Is.EqualTo(2));
```

Comment:
But the count *did* change, which kind of proves that the test is not very clear. Initially we have 1 PKO (added by the first object). Then we update the first object by Id with a list that has a different PKO, which causes the second object to be added to the realm (bumping the count of PKO's to 2), but overwriting the `PrimaryKeyWithPKList`'s list so that it only contains a single item.